lib/internal/mime.js

```diff
@@ -0,0 +1,318 @@
+'use strict';
+
+const {
+  ArrayPrototypeJoin,
+  ObjectCreate,
+  MapIteratorPrototypeNext,
+  RegExpPrototypeTest,
+  RegExpPrototypeExec,
+  StringPrototypeReplace,
+  StringPrototypeToLowerCase,
+  StringPrototypeSlice,
+  StringPrototypeCharAt,
+  SymbolReplace,
+  SymbolIterator,
+  SafeMap: Map,
+} = primordials;
+
+const NotHTTPTokenCodePoint = /[^!#$%&'*+\-.^_`|~A-Za-z0-9]/g;
+const NotHTTPQuotedStringCodePoint = /[^\t\u0020-~\u0080-\u00FF]/g;
+NotHTTPQuotedStringCodePoint[SymbolReplace] = null;
+
+const HTTP_TAB_OR_SPACE = '\t ';
+const HTTP_WHITESPACE = ArrayPrototypeJoin(['\r\n', ...HTTP_TAB_OR_SPACE], '');
+const BEGINNING_WHITESPACE = new RegExp(
+  `^[${StringPrototypeReplace(HTTP_WHITESPACE, /\W/g, '\\$&')}]*`);
```

Comment:
Nit: I'd prefer it like this here and below, IMO it's quite dense already
```suggestion
  '^[' + StringPrototypeReplace(HTTP_WHITESPACE, /\W/g, '\\$&') + ']*');
```

Reply:
any reason to do this rather than use line spacing? i'd rather keep to a single type of string concatenation overall if line spacing would be ok.

Reply:
Could you please clarify this?

> any reason to do this rather than use line spacing?

Perhaps you meant to say line interpolation, then the answer would be that avoiding it clarifies the code a little in this case and makes it easier to distinguish the RegExp-related symbols from the other ones and therefore makes the regexp easier to understand.